### PR TITLE
Automate construction of complete Turnstile ingestion package

### PR DESCRIPTION
**Pull request by glguy**
See #946

Remaining work: incorporate into CI

### CODE REVIEW

**Comment by glguy** [comment]
@tuxji I think what's left is to invoke this script from the CI process and save the zip file output somewhere.

**Review by kquick — APPROVED** [review]

**Comment by tuxji** [comment]
Looks good.  I'm traveling on Wednesday, so I won't be able to make the CI changes until Thursday.

**Comment by tuxji** [comment]
@glguy @kquick Please review the [commit](https://github.com/ge-high-assurance/RACK/pull/960/commits/197cb72f42727022da5cdf81c50b266f3c9a898f) I just added to this pull request (I also rebased it on the master branch to bring it up to date).  I've modified the release workflow to build the Turnstile ingestion package and attach it to the release. I've also taken advantage of the fact that we're now using Ubuntu 22.04 which already has swi-prolog version 8.4.2 to simplify all the places which had to add swi-prolog's ppa repository before.  I figure we can remove these extra steps now as long as ASSIST doesn't need any features that are only in SWI-Prolog version 9.0.4 (the ppa version).

**Comment by tuxji** [comment]
P.S. I ran the workflow's new commands manually in my checkout so I think they should work on the github runner.  However, you should review these [commands](https://github.com/ge-high-assurance/RACK/blob/issue946/.github/workflows/assemble-turnstile-data.yml) as well.  I have to wait until we merge this pull request into the master branch before the Actions tab will let me call the assemble-turnstile-data.yml workflow from the Actions tab.  If it works when I call it from the Actions tab for testing, it should also work when CI calls it for the next release.

File: .github/workflows/assemble-turnstile-data.yml
Comment (kquick) [inline .github/workflows/assemble-turnstile-data.yml:4]: I don't think `workflow_call` should have a trailing colon (https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_call)

**Review by kquick — APPROVED** [review]
Looks good, and it's nice not to have to do extra work to get SWI Prolog available.

I noticed that this new action isn't run on the branch... I suppose we have  to wait until merge time to see if it works OK?

**Comment by kquick** [comment]
> I noticed that this new action isn't run on the branch... I suppose we have to wait until merge time to see if it works OK?

Ah, I re-read your P.S. comment after making this remark... just ignore it.  :-)

File: .github/workflows/assemble-turnstile-data.yml
Comment (tuxji) [reply .github/workflows/assemble-turnstile-data.yml:4]: Yes, you're correct.  I'd copied this file from build-virtual-machine.yml where the `workflow_call` had been followed by an input, and I'd deleted the input but not the trailing colon.  I'll go ahead, merge the pull request, and if CI fails immediately, I'll fix it, otherwise I'll wait for CI to find any other problems that I can fix at the same time too.

**Pull request merged**
